contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -358,6 +322,59 @@ contract RiskFundConverter is AbstractTokenConverter {
         }
     }
 
+    /// @dev Update the reserve of the asset for the specific pool after transferring to risk fund
+    /// and transferring funds to the protocol share reserve
+    /// @param comptroller Comptroller address (pool)
+    /// @param asset Asset address
+    /// @custom:event AssetTransferredToDestination emits when poolsAssetsDirectTransfer is enabled for entered comptroller and asset
+    /// @custom:event AssetsReservesUpdated emits when poolsAssetsDirectTransfer is not enabled for entered comptroller and asset
+    function _updateAssetsState(address comptroller, address asset) internal override returns (uint256) {
+        if (!ensureAssetListed(comptroller, asset)) revert MarketNotExistInPool(comptroller, asset);
+
+        IERC20Upgradeable token = IERC20Upgradeable(asset);
+        uint256 currentBalance = token.balanceOf(address(this));
+        uint256 assetReserve = assetsReserves[asset];
+        if (currentBalance > assetReserve) {
+            uint256 balanceDifference;
+            unchecked {
+                balanceDifference = currentBalance - assetReserve;
+            }
+            if (poolsAssetsDirectTransfer[comptroller][asset]) {
+                uint256 previousDestinationBalance = token.balanceOf(destinationAddress);
+                token.safeTransfer(destinationAddress, balanceDifference);
+                uint256 newDestinationBalance = token.balanceOf(destinationAddress);
+
+                emit AssetTransferredToDestination(comptroller, asset, balanceDifference);
+                IRiskFund(destinationAddress).updatePoolState(
+                    comptroller,
+                    asset,
+                    newDestinationBalance - previousDestinationBalance
+                );
+                return 0;
+            }
+            return balanceDifference;
+        }
+    }
+
+    function _postPrivateConversion(
+        address comptroller,
+        address tokenAddressIn,
+        uint256 convertedTokenInBalance,
+        address tokenAddressOut,
+        uint256 convertedTokenOutBalance
+    ) internal override {
+        if (convertedTokenInBalance > 0) {
+            assetsReserves[tokenAddressIn] += convertedTokenInBalance;
+            poolsAssetsReserves[comptroller][tokenAddressIn] += convertedTokenInBalance;
+            emit AssetsReservesUpdated(comptroller, tokenAddressIn, convertedTokenInBalance);
+        }
```

Comment:
I don't think we have to do this. These funds are not in this contract anymore, they were transferred to the destination contract in the private conversion.

Reply:
```suggestion
        if (convertedTokenInBalance > 0) {
           emit AssetTransferredToDestination(comptroller, tokenAddressIn, convertedTokenInBalance);
            IRiskFund(destinationAddress).updatePoolState(
                comptroller,
                tokenAddressIn,
                convertedTokenInBalance
            );
        }
```
Yes, we should not update the states as this amount is already transferred to destination but we have to call `updatePoolState` for the riskfund to update it's state after transfer.